### PR TITLE
Fix horner and horner2d for complex inputs

### PR DESCRIPTION
**Pull request by rmjarvis**
I just discovered that our horner and horner2d functions had a bug when the inputs are complex.

Basically, it was recasting them to float for no good reason.  I guess maybe thinking to recast int or float32 arrays to float, but I don't think that's actually necessary, so better to not bother about that.  Anyway, fixed here.

### CODE REVIEW

File: galsim/utilities.py [hunk 1]
```diff
@@ -705,13 +705,12 @@ def horner2d(x, y, coefs, dtype=None, triangle=False):
...
-        coefs = np.array(coefs, copy=False, dtype=float)
+        x = np.array(x, copy=False)
+        y = np.array(y, copy=False)
+        coefs = np.array(coefs, copy=False)
 
-    if len(x) != len(y):
-        raise GalSimIncompatibleValuesError("x and y are not the same size",
-                                            x=x, y=y)
+    if x.shape != y.shape:
+        raise GalSimIncompatibleValuesError("x and y are not the same size", x=x, y=y)
```
Comment (jmeyers314) [inline galsim/utilities.py]: s/size/shape

File: galsim/utilities.py [hunk 2]
```diff
@@ -705,13 +705,12 @@ def horner2d(x, y, coefs, dtype=None, triangle=False):
         y = np.ascontiguousarray(y, dtype=float)
         coefs = np.ascontiguousarray(coefs, dtype=float)
     else:
-        x = np.array(x, copy=False, dtype=float)
-        y = np.array(y, copy=False, dtype=float)
-        coefs = np.array(coefs, copy=False, dtype=float)
+        x = np.array(x, copy=False)
```
Comment (jmeyers314) [inline galsim/utilities.py]: I think this change (and the 1d version too) makes the docstrings of `_horner`, `_horner2d` lies.  The dtype can now be complex.  Actually, looks like the docstrings were previous lies too.

Comment (jmeyers314) [reply galsim/utilities.py]: Also, s/coeff/coef for 1d and s/coeff/coefs for 2d.

**Comment by rmjarvis** [comment]
Thanks Josh.  Fixed.

**Pull request merged**